source/_posts/2018-01-03-using-git-hooks-to-improve-your-day-to-day-workflow.markdown

```diff
@@ -0,0 +1,108 @@
+
+{% codeblock lang:bash %}
+#!/usr/bin/env bash
+
+if grep -q -i -e "WIP" -e "work in progress" $1; then
+    read -p "You're about to add a WIP commit, do you want to run the CI? [y|n] " -n 1 -r < /dev/tty
+    echo
+    if echo $REPLY | grep -E '^[Nn]$' > /dev/null; then
+        echo "[skip ci]" >> $1
+    fi
+fi
+{% endcodeblock %}
+
+This example searches the commit message entered by the developer for a certain pattern, if the pattern is found it assumes that the commit is a work in progress code and asks the developer to confirm if he wants to run this code in the CI, if the user says *no* then the hook modifies the commit to add a *[skip ci]* label.
+
+### PRE-PUSH
+
+The `pre-push` hook runs during git push, after the remote refs have been updated but before any objects have been transferred. It receives the name and location of the remote as parameters, and a list of to-be-updated refs through stdin. You can use it to validate a set of ref updates before a push occurs, or to abort a push to a particuar branch, a non-zero exit code will abort the push.
+
+{% codeblock lang:bash %}
+#!/usr/bin/env bash
+
+protected_branch='master'
+current_branch=$(git symbolic-ref HEAD | sed -e 's,.*/\(.*\),\1,')
+
+if [ $protected_branch = $current_branch ]; then
+    read -p "You're about to push master, is that what you intended? [y|n] " -n 1 -r < /dev/tty
+    echo
+    if echo $REPLY | grep -E '^[Yy]$' > /dev/null; then
+        exit 0 # push will execute
+    fi
+    exit 1 # push will not execute
+else
+    exit 0 # push will execute
+fi
+{% endcodeblock %}
+
+This script will ask for confirmation if the developer trys to push to an important branch like *master*, if the developer confirms then the push will happen as usual, otherwise it will prevent the branch from being pushed.
+
+## Setting up hooks in a project
+
+Since the `.git` folder itself is not under version control the hooks are only available to the developers that install them, which for some hooks might be good enougth, but not for others.
+
+For example, if you want to protect a branch for all developers it would be great that this could be under version control and when developers checkout the code they have that available. Well, this can easily be done!
+
+If you have a setup script for you project you can just add `git config core.hooksPath <path-to-hooks-folder>` and this will make you repository look for hooks in that folder which you can have in the repository under version control. After this, when one developers installs the project, they automatically have the hooks in place and working.
```

Comment:
`After this, when one developers installs the project, they automatically have the hooks in place and working.`

=>

`After this, when another developer install the project, the hooks will be in place and working.`

Reply:
well, it can be rewrite in different ways, but the `when one developers installs the project` part should be fixed

Reply:
maybe `when a developer installs the project`